packages/flutter/lib/src/cupertino/text_field.dart

```
@@ -0,0 +1,657 @@
+  void initState() {
+    super.initState();
+    if (widget.controller == null) {
+      _controller = TextEditingController();
+      _controller.addListener(updateKeepAlive);
+    }
+  }
+
+  @override
+  void didUpdateWidget(CupertinoTextField oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller == null && oldWidget.controller != null) {
+      _controller = TextEditingController.fromValue(oldWidget.controller.value);
+      _controller.addListener(updateKeepAlive);
+    }
+    else if (widget.controller != null && oldWidget.controller == null)
+      _controller = null;
+    final bool isEnabled = widget.enabled ?? true;
+    final bool wasEnabled = oldWidget.enabled ?? true;
+    if (wasEnabled && !isEnabled) {
+      _effectiveFocusNode.unfocus();
+    }
+  }
+
+  @override
+  void dispose() {
+    _focusNode?.dispose();
+    _controller?.removeListener(updateKeepAlive);
+    super.dispose();
+  }
+
+  void _requestKeyboard() {
+    _editableTextKey.currentState?.requestKeyboard();
+  }
+
+  RenderEditable get _renderEditable => _editableTextKey.currentState.renderEditable;
+
+  void _handleTapDown(TapDownDetails details) {
+    _renderEditable.handleTapDown(details);
+  }
+
+  void _handleTap() {
+    _renderEditable.handleTap();
+    _requestKeyboard();
+  }
+
+  void _handleTapCancel() {
+  }
+
+  void _handleLongPress() {
+    _renderEditable.handleLongPress();
+  }
+
+  @override
+  bool get wantKeepAlive => _controller?.text?.isNotEmpty == true;
+
+  @override
+  void deactivate() {
+    super.deactivate();
+  }
+
+  bool _shouldShowAttachment({
+    OverlayVisibilityMode attachment,
+    bool hasText,
+  }) {
+    switch (attachment) {
+      case OverlayVisibilityMode.never:
+        return false;
+      case OverlayVisibilityMode.always:
+        return true;
+      case OverlayVisibilityMode.editing:
+        return hasText;
+      case OverlayVisibilityMode.notEditing:
+        return !hasText;
+    }
+    assert(false);
```

assert that attachment is non-null before the switch, then you can remove this assert.

---

not sure if I understood correctly. This is just to make sure adding new enum values in the future fails visibly